### PR TITLE
core/peer: remove deprecated ID.Pretty method

### PR DESCRIPTION
**Pull request by marten-seemann**
This method has been deprecated for more than a year now.

### CODE REVIEW

File: p2p/net/connmgr/decay.go [hunk 1]
```diff
@@ -337,7 +337,7 @@ func (t *decayingTag) Remove(p peer.ID) error {
 	default:
 		return fmt.Errorf(
 			"unable to remove decaying tag for peer %s, tag %s; queue full (len=%d)",
-			p.Pretty(), t.name, len(t.trkr.removeTagCh))
+			p.String(), t.name, len(t.trkr.removeTagCh))
```
Comment (sukunrt) [inline p2p/net/connmgr/decay.go:340]: Can keep this just `p`

**Review by sukunrt — APPROVED** [review]

File: p2p/net/connmgr/decay.go (hunk 1 above)
Comment (marten-seemann) [inline p2p/net/connmgr/decay.go:340]:
```suggestion
			p, t.name, len(t.trkr.removeTagCh))
```

**Comment by marten-seemann** [comment]
Just putting this out there, this will require some amount of cleanup for our users:
* Kubo: https://github.com/search?q=repo%3Aipfs%2Fkubo%20Pretty()&type=code
* Lotus: https://github.com/search?q=repo%3Afilecoin-project%2Flotus%20Pretty()&type=code

I still think we should go ahead with the removal, the whole purpose of deprecation is that we can eventually clean up our API.

**Comment by marten-seemann** [comment]
cc @p-shahi for visibility

**Pull request merged**